src/dataframe/dataframe.jl

```diff
@@ -427,9 +427,23 @@ _onebased_check_error(i, col) =
                         "columns that use 1-based indexing, but " *
                         "column $i has starting index equal to $(firstindex(col))"))
 
-# note: these type assertions are required to pass tests
+"""
+    nrow(df::AbstractDataFrame)
+
+Return the number of rows in an `AbstractDataFrame` `df`.
+
+See also: [`ncol`](@ref), [`size`](@ref).
+
+# Examples
+
+```jldoctest
+julia> df = DataFrame(i=1:10, x=rand(10), y=rand(["a", "b", "c"], 10));
+
+julia> nrow(df)
+10
+```
+""" # note: these type assertions are required to pass tests
 nrow(df::DataFrame) = ncol(df) > 0 ? length(_columns(df)[1])::Int : 0
-ncol(df::DataFrame) = length(index(df))
```

Comment:
Why do you separate `nrow` from `ncol` definitions?

Reply:
Because `ncol` definition is generic and works for any `AbstractDataFrame` so it goes into abstractdataframe/abstractdataframe.jl.
While `nrow` needs a separate definition for `DataFrame` and `SubDataFrame` (and they are respectively in dataframe/dataframe.jl and subdataframe/subdataframe.jl).

This separation is a consequence of your request to attach docsting of `nrow` to actual method definition (and not to `function nrow end` placeholder that was in abstractdataframe/abstractdataframe.jl in my initial version of this PR).

Reply:
(I preferred having `function nrow end` and a docstring in abstractdataframe/abstractdataframe.jl so if you now agree to do this this way I will change the PR)